### PR TITLE
Introduce Task Clustering with creating a new clustered SwarmFlow

### LINKED ISSUE #13: Integrate task clustering with creating a new clustered SwarmFlow
By using the clustered DAG given by #12 , fireworks need to be combined sequentially and parallely.
After combining fireworks, given SwarmFlow needs to be clustered and create new SwarmFlow.

### PR DESCRIPTION
**Pull request by AyeshW**
This PR resolves #13 

It has functionalities to combine multiple firetasks into a single firework sequentially and parallely.

Also it clusters a SwarmFlow identified by a unique sf_id and create a new SwarmFlow with clustered fireworks.

Script Path:
"swarmform/core/cluster.py"

### CODE REVIEW

File: swarmform/core/cluster.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        Update the parent-child relationships after clustering a firework
+        by replacing all the instances of old_id with new_id
+        Args:
+            links_dict (list): Existing parent-child relationship list
+            old_id (int): Existing id of the firework
+            new_id (int): New id of the firework
+
+        Returns:
+            links_dict (list): Updated parent-child relationship list
+        """
```
Comment (KalanaDananjaya) [inline swarmform/core/cluster.py]: wrong indentation level for comment and closing """

File: swarmform/core/cluster.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    # Enumerate child ids and replace it with the new id
+    for parent_id in links_dict:
+        child_id_list = links_dict[parent_id]
+        for index, ID in enumerate(child_id_list):
```
Comment (KalanaDananjaya) [inline swarmform/core/cluster.py]: is it ok to use a variable with capital letters?

File: swarmform/core/cluster.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    # Enumerate parent ids and replace it with the new id
+    if old_id in links_dict:
+        links_dict[new_id] = links_dict.pop(old_id)
+    return links_dict
+
+
+def cluster_sf(swarmpad, sf_id):
+
+    """
+            Pull the swarmflow from given sf_id and create a clustered swarmflow
```
Comment (KalanaDananjaya) [inline swarmform/core/cluster.py]: wrong indentation level for comment

**Review by KalanaDananjaya — APPROVED** [review]

File: swarmform/core/cluster.py (hunk 1 above)
Comment (AyeshW) [reply swarmform/core/cluster.py]: Fixed

File: swarmform/core/cluster.py (hunk 2 above)
Comment (AyeshW) [reply swarmform/core/cluster.py]: Fixed

File: swarmform/core/cluster.py (hunk 3 above)
Comment (AyeshW) [reply swarmform/core/cluster.py]: Fixed

**Pull request merged**